tests/eas/st_hold.py

```diff
@@ -0,0 +1,408 @@
+                raise AssertionError(
+                    "During test run, avg_freq {} was not less than"
+                    " {} lower than max_freq {} on test CPU {}\n".format(
+                        avg_freq, self.frequency_margin, max_freq, self.test_cpu))
+        else:
+            if not max_freq - avg_freq > self.frequency_margin:
+                raise AssertionError(
+                    "During test run, avg_freq {} was not more than"
+                    " {} lower than max_freq {} on test CPU {}\n".format(
+                        avg_freq, self.frequency_margin, max_freq, self.test_cpu))
+
+class RTBoostedTaskTest(_SchedtuneHoldTest):
+    """
+    Schedtune Boost Hold keeps the boost level active for at least
+    50ms since the last enqueue, for qualifying tasks. Eligibility is
+    controlled by a sched_feature. In the default configurations,
+    NO_SCHEDTUNE_BOOST_HOLD_ALL is the sched_feature enabled, and
+    this means that only rt-tasks in boosted groups trigger the
+    boost hold mechanism. If SCHEDTUNE_BOOST_HOLD_ALL is enabled
+    instead, any rt or cfs task in a boosted group will trigger
+    the boost holding.
+    This test uses a 40ms periodic 1% rt task in a 100% boosted
+    group to cause schedutil to raise the cpu frequency to max
+    for the entire duration of the test.
+    """
+    task_name = "1pct_rt_100"
+    cgroup_name = BOOSTED_CGROUP_NAME
+
+    @classmethod
+    def setUpClass(self, *args, **kwargs):
+        super(RTBoostedTaskTest, self).setUpClass(*args, **kwargs)
+
+    @classmethod
+    def populate_params(self):
+        self.rtapp = RTA(self.target,self.rta_name)
+        self.rtapp.conf(
+            kind='profile',
+            params={
+                self.task_name: Periodic(
+                    period_ms=40, duty_cycle_pct=1,
+                    duration_s=2, sched={ 'policy': 'FIFO', },
+                    cpus=self.test_cpu,
+                ).get(),
+            },
+            run_dir='/tmp'
+        )
+
+    def test_cpu_frequency(self):
+        self._test_cpu_frequency(lower_freq_tgt=True)
+
+class RTUnboostedTaskTest(_SchedtuneHoldTest):
+    """
+    Schedtune Boost Hold keeps the boost level active for at least
+    50ms since the last enqueue, for qualifying tasks. Eligibility is
+    controlled by a sched_feature. In the default configurations,
+    NO_SCHEDTUNE_BOOST_HOLD_ALL is the sched_feature enabled, and
+    this means that only rt-tasks in boosted groups trigger the
+    boost hold mechanism. If SCHEDTUNE_BOOST_HOLD_ALL is enabled
+    instead, any rt or cfs task in a boosted group will trigger
+    the boost holding.
+    This test uses a 40ms periodic 1% rt task in a 0% boosted
+    group and verifies that schedutil does not raise the cpu
+    frequency to max for the entire duration of the test.
+    """
+    task_name = "1pct_rt_0"
+    cgroup_name = UNBOOSTED_CGROUP_NAME
+
+    @classmethod
+    def setUpClass(self, *args, **kwargs):
```

Comment:
I haven't checked the exact behavior of nosetest but is that needed ? This code can probably be removed.